### PR TITLE
 feat(dynamic-forms): add custom type support (closes #314)

### LINKED ISSUE #314: [Enhancement] Enable insertion of custom input components using dynamic forms
#### Feature Request
It would work like: 
```
this.elements = [{
      name: 'titleText',
      label: 'Title Text',
      type: MyCustomInputComponent,
      default: this.titleText,
}]
```

I imagine **MyCustomDialComponent** to implement some kind of interface so that the dynamic forms can access and set its values.

@emoralesb05

### PR DESCRIPTION
**Pull request by emoralesb05**
## Description

add the ability to add your own custom type as a dynamic element

this will allow developers to easily add their own components into the dynamic forms component creation by following the instructions in our demo.

### What's included?

- support for custom types

#### Test Steps

- [ ] `npm run serve`
- [ ] Go to http://localhost:4200/#/components/dynamic-forms
- [ ] See last demo on page and README with information about cystom types.

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

closes https://github.com/Teradata/covalent/issues/314

### CODE REVIEW

**Comment by JulieMarie** [comment]
expected

Is this expected behavior with the list staying up after something has been selected?  Just seems a little weird to me

**Comment by emoralesb05** [comment]
Thats how chips work by default without removing the selections from the list

File: src/platform/dynamic-forms/README.md [hunk 1]
```diff
@@ -64,6 +64,8 @@ export interface ITdDynamicElementConfig {
 }
 ```
 
+NOTE: For cystom types, you need to implement a `[control]` property and use it in your underlying element.
```
Comment (jeremysmartt) [inline src/platform/dynamic-forms/README.md]: custom not cystom :)

File: src/platform/dynamic-forms/dynamic-forms.component.spec.ts [hunk 2]
```diff
@@ -324,6 +325,28 @@ describe('Component: TdDynamicForms', () => {
...
+      type: TdDynamicTestComponent,
+      default: 'value',
+    }];
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
+      expect(fixture.debugElement.queryAll(By.directive(TdDynamicElementComponent)).length).toBe(1);
+      let dynamicFormsComponent: TdDynamicFormsComponent =
+          fixture.debugElement.query(By.directive(TdDynamicFormsComponent)).componentInstance;
+      expect(dynamicFormsComponent.valid).toBeTruthy();
+      /* tslint:disable-next-line */
```
Comment (jeremysmartt) [inline src/platform/dynamic-forms/dynamic-forms.component.spec.ts]: why disable lint here?

**Pull request merged**